From f6112e3a942c5f03eb0ce9268644a410f04f2859 Mon Sep 17 00:00:00 2001
From: "Lain \"Fearyncess\" Yang" <fsf@live.com>
Date: Fri, 30 Aug 2024 17:27:37 +0800
Subject: [PATCH 1/2] AOSCOS: r600: disable buggy VC-1 hardware decoding

Per commit 6cd30f5d730c ("radeon/uvd: disable VC-1 simple/main on UVD
2.x"), VC-1 simple/main decoding was broken on at least UVD-1. According
to our testing, VC-1 hardware decoding via VA-API was broken for later UVD
versions as well (on a TeraScale 2-based card).

Disable VC-1 hardware decoding for all r600-driven cards.

Link: https://cgit.freedesktop.org/mesa/mesa/commit/?id=6cd30f5d730c3c23251a8e72e17fe65810460764
Link: https://bugs.freedesktop.org/show_bug.cgi?id=66452

Signed-off-by: Lain "Fearyncess" Yang <fsf@live.com>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/gallium/drivers/r600/radeon_video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/drivers/r600/radeon_video.c b/src/gallium/drivers/r600/radeon_video.c
index 670e0aafb9e..6e5abd8d1a1 100644
--- a/src/gallium/drivers/r600/radeon_video.c
+++ b/src/gallium/drivers/r600/radeon_video.c
@@ -237,7 +237,7 @@ int rvid_get_video_param(struct pipe_screen *screen,
 		case PIPE_VIDEO_FORMAT_MPEG4_AVC:
 			return true;
 		case PIPE_VIDEO_FORMAT_VC1:
-			return true;
+			return false;
 		case PIPE_VIDEO_FORMAT_HEVC:
 			return false;
 		case PIPE_VIDEO_FORMAT_JPEG:
-- 
2.47.1

